### PR TITLE
#162412752admin can change status of incident

### PR DESCRIPTION
**Pull request by bl4ck4ndbr0wn**
#### What does this PR do?
Have CRUD in edit(put resource endpoint to be able to update an existing incident record.
#### Description of Task to be completed?
Test API endpoint user can create a new incident.
 create API endpoint.
#### How should this be manually tested?
Clone the repo as bellow:
 ```
 git clone - b ft-admin-can-change-status-of-incident-162412752 git@github.com: bl4ck4ndbr0wn/iReporter.git
```
Create a virtual environment and activate it
 ```
 python3 - m venv env
 source .env
 ```
 Install all the dependencies needed by
 ```
 pip install - r requirements.txt
 ```
 Run the flask application
 ```
  flask run
 ```
On Postman request either of the two endpoints updating as the api requests.
PATCH  /admin/red-flags/<red-flag-id>
Edit the status of a specific red-flag record.

* :params : {
  “record_type” : String,       // [red-flag, intervention]
  “location” : String,   // Lat Long coordinates
  “status” : String,     // [draft, under investigation, resolved, rejected]
  “Images” : [Image, Image], 
  “Videos” : [Image, Image],
  “comment” : String
}
#### What are the relevant pivotal tracker stories?
 [  # 162412752](https://www.pivotaltracker.com/n/projects/2226962)

### CODE REVIEW

File: app/tests/v2/system/test_records.py [hunk 1]
```diff
@@ -1,5 +1,204 @@
...
+
+    def test_create_new_incident(self):
+        """
+        Test create new incident POST Endpoint.
+        :return: status code 201 (Created)
+        """
+        r = self.create_incident()
+        expected = {"status": 201,
+                    "data": [{
+                        "id": 3,
```
Comment (miritih) [inline app/tests/v2/system/test_records.py]: I like your implementation on tests @bl4ck4ndbr0wn, One thing though, any reason for hardcoding the 'id'?. My guess is, the id will change every time you create a new instance and that means this test case will only pass when the `id=3`.

File: app/tests/v2/system/test_records.py [hunk 2]
```diff
@@ -1,5 +1,204 @@
...
+        r = self.app.patch("/api/v1/red-flags/3/comment",
+                           data=json.dumps(self.update_incident),
+                           headers={"content-type": "application/json",
+                                    "Authorization": f"Bearer {self.token}"})
+
+        self.assertEqual(r.status_code, 202)
+        self.assertDictEqual({"status": 202,
+                              "data": [
+                                    {
+                                        "id": 3,
```
Comment (miritih) [inline app/tests/v2/system/test_records.py]: here too

File: app/tests/v2/system/test_records.py (hunk 1 above)
Comment (miritih) [inline app/tests/v2/system/test_records.py]: I like your implementation on tests @bl4ck4ndbr0wn, One thing though, any reason for hardcoding the 'id'?. My guess is, the id will change every time you create a new instance and that means this test case will only pass when the `id=3`.

File: app/tests/v2/system/test_records.py (hunk 2 above)
Comment (miritih) [inline app/tests/v2/system/test_records.py]: this one too

File: app/tests/v2/system/test_records.py (hunk 1 above)
Comment (bl4ck4ndbr0wn) [reply app/tests/v2/system/test_records.py]: I have implemented tests to drop tables on finishing each test. hence i changed id to 1.

File: app/tests/v2/system/test_records.py (hunk 2 above)
Comment (bl4ck4ndbr0wn) [reply app/tests/v2/system/test_records.py]: I have implemented tests to drop tables on finishing each test. hence i changed id to 1.

**Pull request merged**